src/Compilers/CSharp/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.cs

```diff
@@ -404,7 +400,7 @@ private static bool ReturnsValueWithinExpressionBodiedConstruct(BoundReturnState
                 returnStatement.ExpressionOpt != null &&
                 returnStatement.ExpressionOpt.Syntax != null)
             {
-                SyntaxKind parentKind = returnStatement.ExpressionOpt.Syntax.Parent.Kind();
+                SyntaxKind parentKind = returnStatement.ExpressionOpt.Syntax.Parent!.Kind();
```

Comment:
Same comment applies here (assertion is preferred over suppression)
For reference, here's our annotation guidance: [https://github.com/dotnet/roslyn/blob/main/docs/contributing/Nullable%20Annotations.md#annotation-guidance](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Nullable%20Annotations.md#annotation-guidance)

Reply:
Will change. It'd help to have `RequiredParent` method or property. Was there a discussion about adding such helper?